src/syntax/ppx_sedlex.ml

```diff
@@ -311,16 +523,37 @@ let regexp_of_pattern env =
       | _ ->
           err p.ppat_loc @@ "the " ^ name
           ^ " operator requires two arguments, like " ^ name ^ "(a,b)"
-  and aux p =
+  and aux allow_alias p =
+    let loc = p.ppat_loc in
     (* interpret one pattern node *)
     match p.ppat_desc with
-      | Ppat_or (p1, p2) -> Sedlex.alt (aux p1) (aux p2)
+      | Ppat_or (p1, p2) ->
+          let r1, s1 = aux allow_alias p1 in
+          let r2, s2 = aux allow_alias p2 in
+          if not (StrLocSet.equal s1 s2) then begin
+            let x =
+              try StrLocSet.choose (StrLocSet.diff s1 s2)
+              with Not_found -> StrLocSet.choose (StrLocSet.diff s2 s1)
+            in
+            err loc @@ "variable " ^ x.txt
+            ^ " must occur on both sides of this | pattern"
+          end;
+          (Sedlex.alt r1 r2, s1)
       | Ppat_tuple (p :: pl) ->
-          List.fold_left (fun r p -> Sedlex.seq r (aux p)) (aux p) pl
+          List.fold_left
+            (fun (r1, s1) p ->
+              let r2, s2 = aux allow_alias p in
+              if not (StrLocSet.disjoint s1 s2) then begin
+                let x = StrLocSet.choose (StrLocSet.inter s1 s2) in
```

Comment:
Same here, we could return the full list of duplicated bindings

Reply:
I think it's OK to report one location at a time, just like the OCaml compiler behaves.